tardis/widgets/line_info.py

```diff
@@ -0,0 +1,749 @@
+                .split("\\,")
+            )
+            constituent_units.remove("s^{-1}")
+            constituent_units.insert(
+                constituent_units.index("erg") + 1, "s^{-1}"
+            )
+            constituent_units_string = "\\,".join(constituent_units)
+            unit_in_latex = f"\\mathrm{{{constituent_units_string}}}"
+
+        return f"$\\text{{{label_text}}}\\,[{unit_in_latex}]$"
+
+    @staticmethod
+    def get_middle_half_edges(arr):
+        """
+        Get edges of the middle half range of an array.
+
+        Parameters
+        ----------
+        arr : np.array
+
+        Returns
+        -------
+        list
+        """
+        arr = np.sort(arr)
+        return [
+            (arr[-1] - arr[0]) / 4 + arr[1],
+            (arr[-1] - arr[0]) * 3 / 4 + arr[1],
+        ]
+
+    def plot_spectrum(
+        self,
+        wavelength,
+        luminosity_density_lambda,
+        virt_wavelength,
+        virt_luminosity_density_lambda,
+    ):
+        """
+        Produce a plotly figure widget by plotting the spectrum of model.
+
+        Parameters
+        ----------
+        wavelength : astropy.Quantity
+            Wavelength values of a real spectrum, having unit of Angstrom
+        luminosity_density_lambda : astropy.Quantity
+            Luminosity density lambda values of a real spectrum, having unit
+            of (Hz erg)/Angstrom
+        virt_wavelength : astropy.Quantity
+            Wavelength values of a virtual spectrum, having unit of Angstrom
+        virt_luminosity_density_lambda : astropy.Quantity
+            Luminosity density lambda values of a virtual spectrum, having unit
+            of (Hz erg)/Angstrom
+
+        Returns
+        -------
+        plotly.graph_objects.FigureWidget
+        """
+        initial_zoomed_range = self.get_middle_half_edges(wavelength.value)
+
+        return go.FigureWidget(
+            [
+                go.Scatter(
+                    x=wavelength,
+                    y=luminosity_density_lambda,
+                    name="Real packets",
+                ),
+                go.Scatter(
+                    x=virt_wavelength,
+                    y=virt_luminosity_density_lambda,
+                    name="Virtual packets",
+                ),
+                # Hide a one point scatter trace, to bring boxselect in modebar
+                go.Scatter(
+                    x=wavelength[0],
+                    y=luminosity_density_lambda[0],
+                    mode="markers",
+                    marker=dict(opacity=0),
+                    showlegend=False,
+                ),
+            ],
+            layout=go.Layout(
+                title="Spectrum",
+                xaxis=dict(
+                    title=self.axis_label_in_latex(
+                        "Wavelength", wavelength.unit
+                    ),
+                    exponentformat="none",
+                    rangeslider=dict(visible=True),
```

Comment:
I am assuming that this line is for the plot slider under the spectrum plot. If so, it seems like you do not chose the wavelength range of this plot. Is this correct?
<img width="688" alt="Screen Shot 2020-09-10 at 9 53 45 PM" src="https://user-images.githubusercontent.com/4721877/92842407-85562e00-f3b1-11ea-9453-7076a20498d0.png">

The reason I ask is because the current plot has a wavelength range that extends past the spectrum (the red circle in the picture). There is a range key in rangeslider that might fix this (https://plotly.com/python/reference/layout/xaxis/#layout-xaxis-rangeslider-range). If this isn't an easy fix please don't worry - it's a minor issue.

Reply:
Yes I didn't chose the range. The problem is actually not with rangeslider but because the hidden scatter point which bring box selection lies on rightes edge of plot due to using wavelength[0] and luminosity[0]. Plotly tends to add padding (extra space) around it, hence it looks odd as if it's due to extended range of spectrum. There's an already [open issue](https://github.com/plotly/plotly.js/issues/1775) for it and it's not yet have been worked upon.

Though I've managed to fix it by a workaround i.e. calculating an index which brings this single hidden scatter point in exact middle so that extra padding around has no effect.